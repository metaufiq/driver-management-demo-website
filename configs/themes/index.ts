const themes = {
  colors: {
    primary: '#ff453b'
  },
};

export { themes };