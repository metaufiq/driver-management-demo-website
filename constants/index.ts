export const TOTAL_USER = 30;
export const USER_PER_PAGE = 5;
export const PAGE = TOTAL_USER/USER_PER_PAGE;