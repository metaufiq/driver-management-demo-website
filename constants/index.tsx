import {TiHome} from 'react-icons/ti'
import {RiCalendar2Fill} from 'react-icons/ri'
import {FaUserCircle} from 'react-icons/fa'

export const TOTAL_USER = 30;
export const USER_PER_PAGE = 5;
export const PAGE = TOTAL_USER/USER_PER_PAGE;
export const USER_AVATAR_URL = 'https://randomuser.me/api/portraits/med/men/18.jpg';
export const NAVIGATION_MENU = [
  {
      title: 'Beranda',
      path: '#',
      icon: <TiHome />,
  },
  {
      title: 'Driver Management',
      path: '/driver-management',
      icon: <FaUserCircle />,
      isRoot: true,
  },
  {
      title: 'Pickup',
      path: '#',
      icon: <RiCalendar2Fill />
  },
]