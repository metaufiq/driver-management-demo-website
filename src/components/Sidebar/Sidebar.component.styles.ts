import styled from 'styled-components'

import { MenuItemsProps, StyledComponentProps } from './Sidebar.component.types'

export const Container = styled.div<StyledComponentProps>`
  width: 250px;
  height: 100vh;
  background-color: white;
  @media (max-width: 768px) {
    width: 100%;
    max-height: ${({visible}) => visible ? '200px' : 0};
    transition: all 0.3s ease-out;
  }
`

export const CloseButtonContainer = styled.a`
    display: flex;
    justify-content: end;
    font-size: 0.8rem;
    margin-top: 0.75rem;
    margin-right: 1rem;
    color: ${({theme})=>theme.colors.primary};
    @media (min-width: 768px) {
      display: none;
    }
`

export const MenuItems = styled.li<MenuItemsProps>`
    list-style: none;
    display: flex;
    align-items: center;
    justify-content: start;
    width: 100%;
    padding: 1rem 0;
    border-left: ${({active,theme }) => active ? `4px solid ${theme.colors.primary}` : '0'};
`

export const MenuItemLinks = styled.a`
    display: flex;
    align-items: center;
    padding: 0 2rem;
    font-size: 0.875rem;
    font-weight: bold;
    text-decoration: none;
    color: gray;
    background-color: white;

    &:hover {
        color: ${({theme})=>theme.colors.primary};
    }
`
