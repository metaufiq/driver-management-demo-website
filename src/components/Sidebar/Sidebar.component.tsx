import { ReactElement } from "react";

import { Component } from "../../../index.types";
import { 
  CloseButtonContainer, 
  Container, 
  MenuItemLinks, 
  MenuItems 
} from "./Sidebar.component.styles";
import { Props } from "./Sidebar.component.types";
import { useRouter } from "next/router";

/**
 * Sidebar components
 * @param {Props} props component props 
 * @returns {ReactElement} Sidebar components
 */
const Sidebar:Component<Props> = ({visible, onClose, navigationMenu}):ReactElement=>{
  const router = useRouter()
  const isRootRoute = router.route === '/';
return (
  <Container visible={visible}>
    <CloseButtonContainer
      href="#" 
      onClick={onClose}
    >
        Close
    </CloseButtonContainer>
    {navigationMenu.map((item, index) => {
        return (
            <MenuItems
              key={index} 
              active={
                router.route === item.path ||
                (isRootRoute && item.isRoot!)
              }>
                <MenuItemLinks href={item.path}>
                    {item.icon}
                    <span style={{marginLeft: '16px'}}>{item.title}</span>
                </MenuItemLinks>
            </MenuItems>
        )
    })}
  </Container>
)
}

export default Sidebar