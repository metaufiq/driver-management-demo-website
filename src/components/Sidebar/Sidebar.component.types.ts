import { NAVIGATION_MENU } from "../../constants"

export type MenuItemsProps = {
  active: boolean,
}

export type StyledComponentProps = {
  visible: boolean,
}

export type Props = StyledComponentProps & {
  onClose: VoidFunction,
  navigationMenu: typeof NAVIGATION_MENU
}