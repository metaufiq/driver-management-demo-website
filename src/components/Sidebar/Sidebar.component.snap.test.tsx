import { assertSnapshots } from "@dmd/test";
import Sidebar from ".";
import { Configs } from "@dmd/test/index.types";
import { Props } from "./Sidebar.component.types";
import { themes } from "../../configs/themes";
import { NAVIGATION_MENU } from "../../constants";

jest.mock('next/router', ()=>({
  ...jest.requireActual('next/router'),
  useRouter: ()=>({
    router:{
      route: '/'
    }
  })
}))

describe('NavbarSnapshot', () => {

  const configs: Configs<Props> = [
    {
      desc: 'should render normal Navbar with visible value is equal to true',
      props:{
        visible:true,
        navigationMenu: NAVIGATION_MENU,
        onClose: ()=>{}
      },
      theme: themes,
      useTheme: true
    },
    {
      desc: 'should render normal Navbar with visible value is equal to false',
      props:{
        visible:false,
        navigationMenu: NAVIGATION_MENU,
        onClose: ()=>{}
      },
      theme: themes,
      useTheme: true
    }
  ];

  assertSnapshots(Sidebar, configs);
});
