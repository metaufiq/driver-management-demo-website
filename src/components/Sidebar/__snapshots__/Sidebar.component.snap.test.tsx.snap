// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarSnapshot should render normal Navbar with visible value is equal to false 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ cooIUN"
  >
    <a
      class="sc-gsnTZi pWPJU"
      href="#"
    >
      Close
    </a>
    <li
      class="sc-dkzDqf gwIkhj"
    >
      <a
        class="sc-hKMtZM iOhZzg"
        href="#"
      >
        <svg
          baseProfile="tiny"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          version="1.2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 3s-6.186 5.34-9.643 8.232c-.203.184-.357.452-.357.768 0 .553.447 1 1 1h2v7c0 .553.447 1 1 1h3c.553 0 1-.448 1-1v-4h4v4c0 .552.447 1 1 1h3c.553 0 1-.447 1-1v-7h2c.553 0 1-.447 1-1 0-.316-.154-.584-.383-.768-3.433-2.892-9.617-8.232-9.617-8.232z"
          />
        </svg>
        <span
          style="margin-left: 16px;"
        >
          Beranda
        </span>
      </a>
    </li>
    <li
      class="sc-dkzDqf gwIkhj"
    >
      <a
        class="sc-hKMtZM iOhZzg"
        href="/driver-management"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 496 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
          />
        </svg>
        <span
          style="margin-left: 16px;"
        >
          Driver Management
        </span>
      </a>
    </li>
    <li
      class="sc-dkzDqf gwIkhj"
    >
      <a
        class="sc-hKMtZM iOhZzg"
        href="#"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M17 3h4a1 1 0 0 1 1 1v16a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V4a1 1 0 0 1 1-1h4V1h2v2h6V1h2v2zM4 9v10h16V9H4zm2 2h2v2H6v-2zm5 0h2v2h-2v-2zm5 0h2v2h-2v-2z"
            />
          </g>
        </svg>
        <span
          style="margin-left: 16px;"
        >
          Pickup
        </span>
      </a>
    </li>
  </div>
</DocumentFragment>
`;

exports[`NavbarSnapshot should render normal Navbar with visible value is equal to true 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ iosEpP"
  >
    <a
      class="sc-gsnTZi pWPJU"
      href="#"
    >
      Close
    </a>
    <li
      class="sc-dkzDqf gwIkhj"
    >
      <a
        class="sc-hKMtZM iOhZzg"
        href="#"
      >
        <svg
          baseProfile="tiny"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          version="1.2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 3s-6.186 5.34-9.643 8.232c-.203.184-.357.452-.357.768 0 .553.447 1 1 1h2v7c0 .553.447 1 1 1h3c.553 0 1-.448 1-1v-4h4v4c0 .552.447 1 1 1h3c.553 0 1-.447 1-1v-7h2c.553 0 1-.447 1-1 0-.316-.154-.584-.383-.768-3.433-2.892-9.617-8.232-9.617-8.232z"
          />
        </svg>
        <span
          style="margin-left: 16px;"
        >
          Beranda
        </span>
      </a>
    </li>
    <li
      class="sc-dkzDqf gwIkhj"
    >
      <a
        class="sc-hKMtZM iOhZzg"
        href="/driver-management"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 496 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
          />
        </svg>
        <span
          style="margin-left: 16px;"
        >
          Driver Management
        </span>
      </a>
    </li>
    <li
      class="sc-dkzDqf gwIkhj"
    >
      <a
        class="sc-hKMtZM iOhZzg"
        href="#"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M17 3h4a1 1 0 0 1 1 1v16a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V4a1 1 0 0 1 1-1h4V1h2v2h6V1h2v2zM4 9v10h16V9H4zm2 2h2v2H6v-2zm5 0h2v2h-2v-2zm5 0h2v2h-2v-2z"
            />
          </g>
        </svg>
        <span
          style="margin-left: 16px;"
        >
          Pickup
        </span>
      </a>
    </li>
  </div>
</DocumentFragment>
`;
